### PR TITLE
Add support for components in pods

### PR DESCRIPTION
**Pull request by cs3b**
pods structure among components is pretty popular, this pull request add support for this structure.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 language: node_js
 node_js:
-  - "7"
+  - "8.2.1"
```
Comment (josa42) [inline .travis.yml]: AFAIK Atom is running in node version 6.5.0 and VSCode is running on version 7.4.0, so this change does not make much sense to me. Please revert it and I'll adjust it in the master.

File: main.test.js [hunk 2]
```diff
@@ -1,208 +0,0 @@
-/* eslint-env mocha */
```
Comment (josa42) [inline main.test.js]: While you are completely right, that this file should be removed, it has nothing to do with this PR, so please revert it, I'll remove it in the master.

File: tests/unit/main.test.js [hunk 3]
```diff
@@ -84,6 +93,15 @@ describe('Unit tests', () => {
...
+      assert.deepEqual(detectType('/foo', 'app/components/foo/template.hbs'),           { hostType: 'app', path: 'app/components/foo/template.hbs', part: 'foo', key: 'pod-component-template-hbs' })
+      assert.deepEqual(detectType('/foo', 'app/components/foo/style.scss'),                { hostType: 'app', path: 'app/components/foo/style.scss', part: 'foo', key: 'pod-component-style-scss' })
+      assert.deepEqual(detectType('/foo', 'app/components/foo/style.sass'),                { hostType: 'app', path: 'app/components/foo/style.sass', part: 'foo', key: 'pod-component-style-sass' })
+      assert.deepEqual(detectType('/foo', 'tests/unit/components/foo/component-test.js'),         { hostType: 'app', path: 'tests/unit/components/foo/component-test.js', part: 'foo', key: 'pod-component-unit-js' })
+      assert.deepEqual(detectType('/foo', 'tests/integration/components/foo/component-test.js'),  { hostType: 'app', path: 'tests/integration/components/foo/component-test.js', part: 'foo', key: 'pod-component-integration-js' })
```
Comment (josa42) [inline tests/unit/main.test.js]: Nit picky: Could you adjust the indention?

File: .travis.yml (hunk 1 above)
Comment (cs3b) [inline .travis.yml]: will do

File: main.test.js (hunk 2 above)
Comment (cs3b) [inline main.test.js]: understand, will remove that file

**Comment by cs3b** [comment]
I've apply the feedback. Please let me know if there is anything to improve. :-)

**Comment by cs3b** [comment]
@josa42 Is there anything I can do in order to push it forward?

**Pull request merged**